### ISSUE TITLE
Accessibility issue: Clickable controls should have an ARIA role - anchor tags for eventClick and (date)select 

### ISSUE THREAD

**Issue by shaunnajk** [fullcalendar/fullcalendar#7567]
### Reduced Test Case

https://stackblitz.com/github/fullcalendar/fullcalendar-examples/tree/main/react

### Do you understand that if a reduced test case is not provided, we will intentionally delay triaging of your ticket?

- [X] I understand

### Which connector are you using (React/Angular/etc)?

React

### Bug Description

Clickable date and events need a role.

Issue as reported by our accessibility scanner:
Controls that respond to clicks should be given an ARIA role such as role=button or role=link. div span and *a elements without an href attribute* have no semantic role.
https://www.w3.org/TR/WCAG20-TECHS/F54

Thanks!

![fullCalendar-needs-role](https://github.com/fullcalendar/fullcalendar/assets/12191764/8ae48526-1bef-40a7-987a-3ff1b46a5ae2)

**Comment by arshaw** [comment]
These issues have been greatly improved in [v7.0.0-beta.4](https://github.com/fullcalendar/fullcalendar-workspace/releases/tag/v7.0.0-beta.4).

### +more link
